driver/config/provider_viper.go

```diff
@@ -82,22 +84,31 @@ const (
 	ViperKeyHasherArgon2ConfigParallelism                           = "hashers.argon2.parallelism"
 	ViperKeyHasherArgon2ConfigSaltLength                            = "hashers.argon2.salt_length"
 	ViperKeyHasherArgon2ConfigKeyLength                             = "hashers.argon2.key_length"
+	ViperKeyHasherArgon2ConfigMinimalDuration                       = "hashers.argon2.minimal_duration"
+	ViperKeyHasherArgon2ConfigExpectedDeviation                     = "hashers.argon2.expected_deviation"
+	ViperKeyHasherArgon2ConfigDedicatedMemory                       = "hashers.argon2.dedicated_memory"
 	ViperKeyPasswordMaxBreaches                                     = "password.max_breaches"
 	ViperKeyIgnoreNetworkErrors                                     = "password.ignore_network_errors"
 	ViperKeyVersion                                                 = "version"
-	Argon2DefaultMemory                                      uint32 = 4 * 1024 * 1024
-	Argon2DefaultIterations                                  uint32 = 4
+	Argon2DefaultMemory                                             = uint32(512 * bytesize.MB / bytesize.KB)
+	Argon2DefaultIterations                                  uint32 = 1
 	Argon2DefaultSaltLength                                  uint32 = 16
 	Argon2DefaultKeyLength                                   uint32 = 32
+	Argon2DefaultDuration                                           = 500 * time.Millisecond
```

Comment:
I think 500ms is too long, it should be more in the 200 range:

```suggestion
	Argon2DefaultDuration                                           = 200 * time.Millisecond
```

Reply:
Libsodium (and most other sources) recommend up to 1s. This only affects login requests, so I wanted to stick with that (median 500ms + acceptable deviation 500ms => 0.5-1s).

Reply:
I see, I thought this would be the minimum duration of a password hashing attempt

Reply:
The wording can definitely be improved around that, I will check it again.